projects/core/src/main/java/dan200/computercraft/core/apis/http/websocket/WebsocketHandle.java

```diff
@@ -79,7 +81,21 @@ public final void send(Coerced<String> message, Optional<Boolean> binary) throws
         if (binary.orElse(false)) {
             websocket.sendBinary(LuaValues.encode(text));
         } else {
-            websocket.sendText(text);
+            var data = text;
+            if (websocket.isBinary()) {
+                // Try to convert the string from UTF-8 bytes to UTF-16 codepoints.
+                // If this fails, fall back to normal ANSI.
+                try {
+                    var buf = LuaValues.encode(text);
+                    var bytes = new byte[buf.capacity()];
+                    buf.get(bytes);
+                    data = new String(bytes, StandardCharsets.UTF_8);
```

Comment:
It might be better to use a `CharsetDecoder` here instead, and then fail on `CharacterCodingException` errors, instead of replacing invalid characters with U+FFFD. Not sure, leave as is, will have a think about it.